tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -238,29 +238,28 @@ def class_host_ft(
         Later in tests this host will be used to perform assertions
         """
 
-        skip_yum_update_during_provisioning(template='Kickstart default finish')
-        host = entities.Host(
-            architecture=default_architecture,
-            build=True,
-            compute_resource=module_azurerm_cr,
-            compute_attributes=self.compute_attrs,
-            interfaces_attributes=self.interfaces_attributes,
-            domain=module_domain,
-            organization=module_org,
-            operatingsystem=default_os,
-            location=module_location,
-            name=self.hostname,
-            provision_method='image',
-            image=module_azurerm_finishimg,
-            root_pass=gen_string('alphanumeric'),
-            environment=module_puppet_environment,
-            puppet_proxy=default_smart_proxy,
-            puppet_ca_proxy=default_smart_proxy,
-        ).create()
-        yield host
-        skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
-        with satellite_setting('destroy_vm_on_host_delete=True'):
-            host.delete()
+        with skip_yum_update_during_provisioning(template='Kickstart default finish'):
+            host = entities.Host(
+                architecture=default_architecture,
+                build=True,
+                compute_resource=module_azurerm_cr,
+                compute_attributes=self.compute_attrs,
+                interfaces_attributes=self.interfaces_attributes,
+                domain=module_domain,
+                organization=module_org,
+                operatingsystem=default_os,
+                location=module_location,
+                name=self.hostname,
+                provision_method='image',
+                image=module_azurerm_finishimg,
+                root_pass=gen_string('alphanumeric'),
+                environment=module_puppet_environment,
+                puppet_proxy=default_smart_proxy,
+                puppet_ca_proxy=default_smart_proxy,
+            ).create()
+            yield host
+            with satellite_setting('destroy_vm_on_host_delete=True'):
```

Comment:
if this is getting repeated it can go in the fixture as well ?

Reply:
@omkarkhatavkar  the full hostname there is different every time and is being initiated per class. That can be a good improvement but I would like to keep that separate from this PR.